persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/NonBlockingSoftIndexFileStore.java

```diff
@@ -359,32 +371,29 @@ protected boolean isIndexLoaded() {
    }
 
    @Override
-   public synchronized void stop() {
-      try {
-         logAppender.stopOperations();
-         logAppender = null;
-         compactor.stopOperations();
-         compactor = null;
-         index.stopOperations();
-         index = null;
-         fileProvider.stop();
-         fileProvider = null;
-         temporaryTable = null;
-         indexQueue = null;
-         storeQueue = null;
-      } catch (InterruptedException e) {
-         Thread.currentThread().interrupt();
-         throw log.interruptedWhileStopping(e);
-      } finally {
-         started = false;
-      }
+   public CompletionStage<Void> stop() {
+      return blockingManager.runBlocking(() -> {
+         try {
+            logAppender.stop();
+            compactor.stopOperations();
+            compactor = null;
+            index.stopOperations();
+            index = null;
+            fileProvider.stop();
+            fileProvider = null;
+            temporaryTable = null;
+            indexQueue = null;
+         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw log.interruptedWhileStopping(e);
+         }
+      }, "soft-index-stop");
    }
 
-   @Override
+   // TODO: this is probably broken, look into
```

Comment:
?

Reply:
The destroy method is never even invoked, even before my changes. I think i will just remove the method.